web/src/reducers/activities.selectors.js

```diff
@@ -30,31 +30,131 @@ export const selectAllActivities = ({
 const selectBudgetForActivity = ({ budget }, { aKey }) =>
   budget.activities[aKey];
 
-export const makeSelectCostAllocateFFP = () =>
-  createSelector(
-    [selectActivityByIndex, selectBudgetForActivity],
-    (activity, budget) => {
-      const { costAllocation } = activity;
-      const { costsByFFY } = budget;
-
-      const byYearData = Object.entries(costsByFFY)
-        .filter(([year]) => year !== 'total')
-        .map(([year, costs]) => {
-          const { ffp } = costAllocation[year];
-          const ffpSelectVal = `${ffp.federal}-${ffp.state}`;
-
-          return {
-            year,
-            total: costs.total,
-            medicaidShare: costs.medicaidShare,
-            ffpSelectVal,
-            allocations: { federal: costs.federal, state: costs.state }
-          };
-        });
+export const selectCostAllocationForActivityByIndex = createSelector(
+  selectActivityByIndex,
+  ({ costAllocation }) => costAllocation
+);
 
-      return { byYearData, costAllocation };
-    }
-  );
+export const selectActivityCostSummary = createSelector(
+  selectApdYears,
+  selectActivityByIndex,
+  (
+    // This intermediate selector gets the budget for an activity by index
+    // rather than by key.
+    {
+      apd: {
+        data: { activities }
+      },
+      budget
+    },
+    { activityIndex }
+  ) => budget.activities[activities[activityIndex].key],
+  (
+    // This intermediate selector maps key personnel into the same data
+    // structure as activity state personnel IF AND ONLY IF we are working
+    // on the first activity. Key personnel get rolled into the first activity,
+    // Program Administration. For other activities, just returns empty arrays.
+    {
+      apd: {
+        data: { keyPersonnel, years }
+      }
+    },
+    { activityIndex }
+  ) =>
+    years.reduce(
+      (o, ffy) => ({
```

Comment:
What is `o`? Could we maybe add a comment?

Reply:
It's the accumulator object for the reducer. I never know what to call it... It doesn't have an inherent name, it's just the temporary thing being built up. But I guess in this case, it's the key personnel, so that seems like a good and decent name for it.